dkg/pedersen/dkg.go

```diff
@@ -0,0 +1,216 @@
+package pedersen
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/dedis/cothority"
+	"github.com/dedis/kyber"
+	"github.com/dedis/kyber/util/key"
+	"github.com/dedis/onet"
+	"github.com/dedis/onet/log"
+
+	dkgpedersen "github.com/dedis/kyber/share/dkg/pedersen"
+)
+
+// Name is the protocol identifier string.
+const Name = "Pedersen_DKG"
+
+func init() {
+	onet.GlobalProtocolRegister(Name, NewSetup)
+}
+
+// Setup can give the DKG that can be used to get the shared public key.
+type Setup struct {
+	*onet.TreeNodeInstance
+	DKG       *dkgpedersen.DistKeyGenerator
+	Threshold uint32
+	Finished  chan bool
+	Wait      bool
+
+	nodes   []*onet.TreeNode
+	keypair *key.Pair
+	publics []kyber.Point
+
+	structStartDeal chan structStartDeal
+	structDeal      chan structDeal
+	structResponse  chan structResponse
+	structWaitSetup chan structWaitSetup
+	structWaitReply chan []structWaitReply
+}
+
+// NewSetup initialises the structure for use in one round
+func NewSetup(n *onet.TreeNodeInstance) (onet.ProtocolInstance, error) {
+	o := &Setup{
+		TreeNodeInstance: n,
+		keypair:          key.NewKeyPair(cothority.Suite),
+		Finished:         make(chan bool, 1),
+		Threshold:        uint32(len(n.Roster().List) - (len(n.Roster().List)-1)/3),
+		nodes:            n.List(),
+	}
+
+	err := o.RegisterHandlers(o.childInit, o.rootStartDeal)
+	if err != nil {
+		return nil, err
+	}
+	err = o.RegisterChannels(&o.structStartDeal, &o.structDeal, &o.structResponse,
+		&o.structWaitSetup, &o.structWaitReply)
+	if err != nil {
+		return nil, err
+	}
+	o.publics = make([]kyber.Point, len(o.nodes))
+	return o, nil
+}
+
+// SharedSecret returns the necessary information for doing shared
+// encryption and decryption.
+func (o *Setup) SharedSecret() (*SharedSecret, error) {
+	return NewSharedSecret(o.DKG)
+}
+
+// NewSharedSecret takes an initialized DistKeyGenerator and returns the
+// minimal set of values necessary to do shared encryption/decryption.
+func NewSharedSecret(gen *dkgpedersen.DistKeyGenerator) (*SharedSecret, error) {
+	if gen == nil {
+		return nil, errors.New("no valid dkg given")
+	}
+	dks, err := gen.DistKeyShare()
+	if err != nil {
+		return nil, err
+	}
+	return &SharedSecret{
+		Index:   dks.Share.I,
+		V:       dks.Share.V,
+		X:       dks.Public(),
+		Commits: dks.Commits,
+	}, nil
+}
+
+// Start sends the Announce-message to all children
+func (o *Setup) Start() error {
+	log.Lvl3("Starting Protocol")
+	// 1a - root asks children to send their public key
+	errs := o.Broadcast(&Init{Wait: o.Wait})
+	if len(errs) != 0 {
+		return fmt.Errorf("boradcast failed with error(s): %v", errs)
```

Comment:
broadcast

Reply:
```suggestion
		return fmt.Errorf("broadcast failed with error(s): %v", errs)
```

Reply:
Perhaps like that you can just click and accept in Github? It's a new feature, unclear how it works...

Reply:
wow that's a useful feature